cmd/erasure-metadata.go

```diff
@@ -468,7 +470,10 @@ func listObjectParities(partsMetadata []FileInfo, errs []error) (parities []int)
 		}
 		// Delete marker or zero byte objects take highest parity.
 		if metadata.Deleted || metadata.Size == 0 {
-			parities[index] = len(partsMetadata) / 2
+			parities[index] = totalShards / 2
+		} else if metadata.TransitionStatus == lifecycle.TransitionComplete {
+			// For tiered objects, read quorum is N/2+1 to ensure simple majority on xl.meta. It is not equal to EcM because the data integrity is entrusted with the warm tier.
+			parities[index] = totalShards - (totalShards/2 + 1)
```

Comment:
This formula seems to be weird for me, it looks like it can be less than N/2, I do not think that is expected, totalShards/2+1 seems to be enough to me